sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -25,6 +25,23 @@ namespace Azure { namespace Core {
     static constexpr IntervalType WindowsToPosixOffsetSeconds = 11644473600LL;
 
   public:
+    /**
+     * @brief Defines the format applied to the fraction part from any @DateFormat
+     *
+     */
+    enum class TimeFractionFormat
+    {
+      /// Decimals are not included when there are no decimals in the source Datetime and any zeros
+      /// from the right are also removed.
+      DropTrailingZeros,
+
+      /// Decimals are included for any Datetime.
```

Comment:
The details of such enum values need to be documented a lot heavily, including edge cases which help the user understand when they would use which format.

Reply:
Some questions that the docs should answer:
- How many maximum digits are supported?
  - What happens if the input string as more digits than what is supported? Will those digits be truncated, rounded, or retained?
- Would "all digits" print all zeros when there is no fractional digit?